Code/Source/RobotControl/Controllers/AckermannController/AckermannControlComponent.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "AckermannControlComponent.h"
+#include "VehicleDynamics/VehicleInputControlBus.h"
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+#include <AzFramework/Physics/RigidBodyBus.h>
+
+namespace ROS2
+{
+    void AckermannControlComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<AckermannControlComponent, AZ::Component>()->Version(1);
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<AckermannControlComponent>("Ackermann Control", "Relays Ackermann commands to vehicle inputs")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
```

Comment:
Maybe add a category as well?
```cpp
->Attribute(AZ::Edit::Attributes::Category, "ROS2")
```

Reply:
This is a good idea - but also applies to all of our components. I created an [issue](https://github.com/RobotecAI/o3de-ros2-gem/issues/223) for this.